File: core/UniqueMorseCodeWords/UniqueMorseCodeWords.py
from typing import List


def uniqueMorseRepresentations(words: List[str]) -> int:
    morse = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---",
             ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
    word_to_morse_map = {chr(index + 97): item for index, item in enumerate(morse)}
    res = {}
    for word in words:
        word_morse = ""
        for ch in word:
            word_morse += word_to_morse_map[ch]
        res[word_morse] = res.get(word_morse, 0) + 1

    return len(res)


if __name__ == '__main__':
    print(uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]))

File: core/UniqueMorseCodeWords/UniqueMorseCodeWords.java
package com.dbc;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class UniqueMorseCodeWords {
    public int uniqueMorseRepresentations(String[] words) {
        String[] morse = new String[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Map<Integer, String> wordToMorse = new HashMap<>();
        for (int i = 0; i < morse.length; i++) {
            wordToMorse.put(i, morse[i]);
        }
        Set<String> res = new HashSet<>();
        for (String word : words) {
            StringBuffer wordMorse = new StringBuffer();
            for (int i = 0; i < word.length(); i++) {
                wordMorse.append(wordToMorse.get(word.charAt(i) - 'a'));
            }
            res.add(wordMorse.toString());
        }
        return res.size();
    }
}
